File: java-solution/src/main/java/leetcode/Solution151.java
package leetcode;

public class Solution151 {
    public String reverseWords(String s) {
        String[] words = s.split("\\s");
        String[] newWord = new String[words.length];
        int c = 0;
        for (int i = words.length - 1; i >= 0; i--) {
            newWord[c] = words[i];
            c++;
        }

        StringBuilder strBuilder = new StringBuilder();
        for (String s1 : newWord) {
            if(s1.equals(" ") || s1.equals("")) continue;
            strBuilder.append(s1);
            strBuilder.append(" ");
        }
        return strBuilder.substring(0, strBuilder.length()-1);
    }
}

File: go-lang/leetcode/solution_151.go
package leetcode

import (
	"strings"
)

func ReverseWords(s string) string {
	words := strings.Split(s, " ")
	newWords := make([]string, len(words))
	c := 0
	for i := len(words) - 1; i >= 0; i-- {
		if words[i] == "" || words[i] == " " {
			continue
		}
		newWords[c] = words[i]
		c++
	}
	return strings.TrimRight(strings.Join(newWords, " "), " ")
}
